ds.go

```diff
@@ -523,7 +520,17 @@ func decryptCredential(cred *Credential, encContext *EncryptionContextValue) (*D
 	}
 
 	dk, err := DecryptDataKey(wrappedKey, encContext)
-
+	if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
I would love it if you could use a case statement instead of two ifs.

Not sure about overriding awserr but I like where your going with adding more info.

cheers